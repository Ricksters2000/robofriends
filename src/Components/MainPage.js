import React, {useState, useEffect, Component} from 'react';
import CardList from '../Components/CardList';
import SearchBox from './SearchBox';
import Scroll from '../Components/Scroll';
import ErrorBoundry from '../Components/ErrorBoundry';
import Header from '../Components/Header';
import './MainPage.css'

class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }

    filterRobots = (robots) => robots.filter(obj => obj.name.toLowerCase().includes(this.props.searchField.toLowerCase()));

    render() {
        const {robots, onSearchChange, isPending} = this.props;

        if(isPending) {
            return <h1>Loading</h1>
        } else {
            return (
                <div className='tc'>
                    <Header />
                    <SearchBox searchChange={onSearchChange} />
                    <Scroll>
                        <ErrorBoundry>
                            <CardList robots={this.filterRobots(robots)} />
                        </ErrorBoundry>
                    </Scroll>
                </div>
            )
        }
    }
}

export default MainPage