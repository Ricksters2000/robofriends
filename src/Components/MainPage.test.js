import React from 'react';
import {shallow} from 'enzyme';
import MainPage from './MainPage';

const mockRobots = [
    {
        name: 'e',
        id: 1,
        email: 'e@gmail.com'
    },
    {
        name: 'hi',
        id: 2,
        email: 'someHi@gmail.com'
    },
    {
        name: 'hello',
        id: 2,
        email: 'someHello@gmail.com'
    },
    {
        name: 'someting',
        id: 2,
        email: 'something@gmail.com'
    }
]

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps} />)
})

it('renders main page', () => {
    expect(wrapper).toMatchSnapshot();
})

it('filters robots correctly', () => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: mockRobots,
        searchField: 'h',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps} />);
    expect(wrapper2.instance().filterRobots([])).toEqual([]);
    expect(wrapper2.instance().filterRobots(mockRobots)).toEqual([
        {
            name: 'hi',
            id: 2,
            email: 'someHi@gmail.com'
        },
        {
            name: 'hello',
            id: 2,
            email: 'someHello@gmail.com'
        },
    ])
})

it('test loading page', () => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: mockRobots,
        searchField: 'h',
        isPending: true
    }

    const wrapper2 = shallow(<MainPage {...mockProps} />);
    expect(wrapper2).toMatchSnapshot();
})