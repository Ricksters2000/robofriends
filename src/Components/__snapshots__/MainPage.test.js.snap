// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main page 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;

exports[`test loading page 1`] = `
<h1>
  Loading
</h1>
`;
